File: epi_judge_java/epi/BuyAndSellStock.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

public class BuyAndSellStock {

    @EpiTest(testDataFile = "buy_and_sell_stock.tsv")
    public static double computeMaxProfit(List<Double> prices) {
        if (prices.isEmpty()) {
            return 0.0;
        }
        var lowestPriceSoFar = prices.getFirst();
        var result = 0.0;

        for (Double price : prices) {
            lowestPriceSoFar = Math.min(lowestPriceSoFar, price);
            result = Math.max(result, price - lowestPriceSoFar);
        }

        return result;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "BuyAndSellStock.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_python/buy_and_sell_stock.py
from typing import List

from test_framework import generic_test


def buy_and_sell_stock_once(prices: List[float]) -> float:
    cur_min = prices[0]
    result = 0.0

    for price in prices:
        if price < cur_min:
            cur_min = price
        result = max(result, price - cur_min)

    return result


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('buy_and_sell_stock.py',
                                       'buy_and_sell_stock.tsv',
                                       buy_and_sell_stock_once))
